tasks/Linux/fetch/sapmachine-fallback.yml

```diff
@@ -0,0 +1,38 @@
+---
+- name: 'Fetch version page'
+  uri:
+    url: 'https://sap.github.io/SapMachine/latest/{{ java_major_version }}'
+    return_content: true
+    follow_redirects: all
+  register: version_page
+
+- name: Find release version
+  set_fact:
+    full_version: >-
+      {{ version_page.content | regex_findall('(1.*[.|-]\d)') +
+          version_page.content | regex_findall('(1.*[.|\d]\d)') }}
+
+- name: 'Fetch artifact checksum file'
+  uri:
+    url: "{{ sapmachine_root_page }}{{ java_distribution }}-{{ full_version[0] }}/\
+      {{ java_distribution }}-{{ java_package }}-{{ full_version[-1] | replace('%2B','.') }}\
+      _linux-x64_bin.sha256.txt"
+    return_content: true
+  register: artifact_checksum_file
+
+- name: Find artifact checksum
+  set_fact:
+    artifact_checksum: >-
+      {{ artifact_checksum_file.content | regex_search('(^\w*)') }}
+
+- name: 'Download artifact'
+  get_url:
+    url: "{{ sapmachine_root_page }}{{ java_distribution }}-{{ full_version[0] }}/\
```

Comment:
Isn't it possible to get download url from root page?

Reply:
The download link is on the git page.
Due to the minor version, there are problems with determining the page of the latest version.

Reply:
How about https://github.com/SAP/SapMachine/releases/latest

Reply:
It give the latest release without considering the version.

Reply:
The latest release for any SapMachine major version can be found at https://sap.github.io/SapMachine/latest/#MAJOR

Reply:
When I tried this, ansible could not redirect the page. So that, I take the full version (with minor) from this page. And then I download exactly by the link.
Knowing the full version, i can go to the page and perform parsing. But is it good idea parsing the page again?